ruby_on_rails/testing/model_testing.md

```diff
@@ -0,0 +1,136 @@
+### Introduction
+
+The last lesson covered unit testing and now we will move into model testing. Here you will get an idea of how to test within Rails for model testing as well as a deeper dive of how to structure these tests.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are model tests?
+*   How do I run a model test?
+*   How to structure model testing?
+
+## What are model tests?
+
+A model test is a unit test in the context of Ruby on Rails. Model tests in this case are designed to test the business logic of your models. Let's say you need to test a custom validation method for your blog post. In this case you would need to reach into the model aspect of your testing skillset. Before you write the RSpec there will be a few tools you will need to familiarize yourself with.
+
+## Setting the test
+
+The first tool that we are going to use to set up the data is [factory bot](https://github.com/thoughtbot/factory_bot). Factory bot is a gem that you can use to set up test data and create necessary data for your tests. Adding Factory bot is similar to adding other testing gems. Add the following line to the to the following section in your Gemfile.
+
+~~~
+
+group :development, :test do
+
+  gem "factory_bot_rails"  #Add this line.
+
+end
+
+~~~
+
+Then in your terminal enter `bundle install` to the terminal where it will install the gem. Now we need to move into setting up our test configuration to utilize Factory bot. First were going to navigate to `spec/support/factory_bot.rb` and add the following configuration. 
+
+~~~
+
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
+
+~~~
+
+Make sure to require this file in `rails_helper.rb`. We will briefly cover what a factory will look like and some things to keep in mind when creating your data. While you can require the file by adding `require support/factory_bot` to the top of the file. Rails also provides this in the commented out line `Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }` which enables you to load all of the support files when booting up. This is useful for not having to manually add all of the support files and avoids issues if you forget to add a file for example. The last thing you might be wondering is how the test database clears itself in between it's tests. Rail's does this automatically for you, however if Rail's is not doing it for you make sure your following the instructions in the rails_helper.rb file and make sure to ask in the [Discord](https://discord.com/channels/505093832157691914/690591236922409012)! There is also a alternative option in the additional resources section if need be for database cleaning.
```

Comment:
I believe this paragraph has two different subjects, so it needs to be broken into different paragraphs. In addition, since we are keeping database cleaner in the lesson, I think I found a great resource that bridges the gap. Though I am open to other suggestions on this. 

```suggestion
You will need to require this file in `rails_helper.rb`. While you can require the file by adding `require support/factory_bot` to the top of the file. Rails also provides this in the commented out line `Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }` which enables you to load all of the support files when booting up. This is useful for not having to manually add all of the support files and avoids issues if you forget to add a file for example. 

Since Factory bot creates data in the test database, you might be wondering if we need to clear the database between tests. In a lot of cases, Rails will automatically do this for you. However, as you develop more complex applications, you might run into situations where data is not being cleared. If you would like to learn more about these situations, the creator of Factory bot has a [great article about using a Database Cleaner](https://thoughtbot.com/blog/how-we-test-rails-applications#database-cleaner). If you run into any issues with this, be sure to ask in our [Discord](https://discord.com/channels/505093832157691914/690591236922409012)!
```